express-notes-api/index.js

```diff
@@ -0,0 +1,87 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json')
+const notes = data.notes;
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  for (const x in notes){
+    arr.push(notes[x]);
+  }
+  res.status(200).json(arr);
+})
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
+  if(id <= 0){
+    const error400 = {error: "id must be a positive integer"}
+    res.status(400).json(error400);
+  }
+  else if(id in notes) {
+    res.json(notes[id]);
+  }
+  else{
+    const error404 = {error: `cannot find note with id ${id}`};
+    res.status(404).json(error404);
+  }
+})
+
+app.post('/api/notes', (req, res) => {
+  if(req.body.content.length == 0){
```

Comment:
You don't want to check the length on content, if there is no content this will throw an error. You can't assume content will be set at all. You just need to check if content is truthy or falsy.

```JAVASCRIPT
if(req.body.content) {
  // Create note
 } else {
  // Send error
}
  ```